crates/evm/evm/src/executors/fuzz/mod.rs

```diff
@@ -98,7 +102,17 @@ impl FuzzedExecutor {
         let max_traces_to_collect = std::cmp::max(1, self.config.gas_report_samples) as usize;
         let show_logs = self.config.show_logs;
 
+        // Start a timer if timeout is set.
+        let start_time = start_timer(self.config.timeout);
+
         let run_result = self.runner.clone().run(&strategy, |calldata| {
+            // Check if the timeout has been reached.
+            if let Some((start_time, timeout)) = start_time {
+                if start_time.elapsed() > timeout {
```

Comment:
this is invoked per run, right?

this adds a clock syscall per run, we can avoid this if we introduce a background task that toggles an atomic (`Arc<AtomicBool>`) instead.

this should be a lot cheaper, we can even spawn this task somewhere higher and keep a handle to communicate or something like

```rust
struct Timeout(Arc<AtomicBool>)
```